osbenchmark/cloud_providers.py

```diff
@@ -0,0 +1,226 @@
+        self.aws_metrics_log_in_dict['metrics_aws_access_key_id'] = metrics_aws_access_key_id
+        self.aws_metrics_log_in_dict['metrics_aws_secret_access_key'] = metrics_aws_secret_access_key
+        self.aws_metrics_log_in_dict['metrics_aws_session_token'] = metrics_aws_session_token
+        self.aws_metrics_log_in_dict['metrics_aws_service'] = metrics_aws_service
+        self.aws_metrics_log_in_dict['metrics_aws_region'] = metrics_aws_region
+
+    def update_client_options_for_metrics(self, client_options):
+        # add options for aws user login:
+        # pass in aws access key id, aws secret access key, aws session token, service and region on command
+        if self.aws_metrics_log_in_dict['metrics_aws_log_in_choice'] is not None:
+            client_options["amazon_aws_log_in"] = 'client_option'
+            client_options["aws_access_key_id"] = self.aws_metrics_log_in_dict['metrics_aws_access_key_id']
+            client_options["aws_secret_access_key"] = self.aws_metrics_log_in_dict['metrics_aws_secret_access_key']
+            client_options["service"] = self.aws_metrics_log_in_dict['metrics_aws_service']
+            client_options["region"] = self.aws_metrics_log_in_dict['metrics_aws_region']
+
+            if self.aws_metrics_log_in_dict['metrics_aws_session_token']:
+                client_options["aws_session_token"] = self.aws_metrics_log_in_dict['metrics_aws_session_token']
+
+        return client_options
+
+    def create_client(self, hosts):
+        credentials = Credentials(access_key=self.aws_log_in_dict["aws_access_key_id"],
+                                  secret_key=self.aws_log_in_dict["aws_secret_access_key"],
+                                  token=self.aws_log_in_dict["aws_session_token"])
+        aws_auth = opensearchpy.Urllib3AWSV4SignerAuth(credentials, self.aws_log_in_dict["region"],
+                                                self.aws_log_in_dict["service"])
+        return opensearchpy.OpenSearch(hosts=hosts, use_ssl=True, verify_certs=True, http_auth=aws_auth,
+                                       connection_class=opensearchpy.Urllib3HttpConnection)
+
+
+    def create_async_client(self, hosts, client_options, client_class):
+        credentials = Credentials(access_key=self.aws_log_in_dict["aws_access_key_id"],
+                            secret_key=self.aws_log_in_dict["aws_secret_access_key"],
+                            token=self.aws_log_in_dict["aws_session_token"])
+        aws_auth = opensearchpy.AWSV4SignerAsyncAuth(credentials, self.aws_log_in_dict["region"],
+                                                     self.aws_log_in_dict["service"])
+        return client_class(hosts=hosts,
+                                        connection_class=async_connection.AsyncHttpConnection,
+                                        use_ssl=True, verify_certs=True, http_auth=aws_auth,
+                                        **client_options)
```

Comment:
Can these be combined into a single function with `async` as a parameter?  Why is the `aws_auth` computation different?  Finally, does the sync client not support client options, or is that handled differently?